### PR TITLE
Display wall/history dates as distances in words

### PR DESCRIPTION
**Pull request by ReneCode**
As Nick suggested I used the data-fns package to get the "distanceInWordsToNow".
For the spec test I was not sure how to mock the 'i18n/locale' getter - so I used the default behavior.

### CODE REVIEW

File: src/components/General/DateAsWords.vue [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<template>
+  <div>{{dateInWords(date)}}</div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import distanceInWordsToNow from 'date-fns/distance_in_words_to_now'
+
+const locales = {
+  en: require('date-fns/locale/en'),
```
Comment (nicksellen) [inline src/components/General/DateAsWords.vue]: can you try loading the locales on-demand? we try and keep the bundle sizes as small as possible.

we do it for the app transations here ---> https://github.com/yunity/karrot-frontend/blob/master/src/store/plugins/i18n.js#L10

File: src/components/General/DateAsWords.vue [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<template>
+  <div>{{dateInWords(date)}}</div>
```
Comment (nicksellen) [inline src/components/General/DateAsWords.vue]: A tooltip showing the full date would be great too

Comment (nicksellen) [reply src/components/General/DateAsWords.vue]: (one of these works nicely http://quasar-framework.org/components/tooltip.html)

**Comment by nicksellen** [comment]
Looks great! Just a couple of comments.

File: package.json [hunk 3]
```diff
@@ -37,6 +37,7 @@
     "babel-plugin-dynamic-import-node": "^1.1.0",
     "babel-runtime": "^6.25.0",
     "clone-deep": "^1.0.0",
+    "date-fns": "^1.29.0",
```
Comment (tiltec) [inline package.json]: Remember to add `yarn.lock` too before merging

File: src/components/General/DateAsWords.spec.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+})
+
+import DateAsWords from './DateAsWords'
+import { mountWithDefaults } from '>/helpers'
+import store from '@/store'
+
+describe('DateAsWords', () => {
+  it('renders 1 day ago', () => {
+    const now = new Date('2017-08-12T12:00:00Z')
+    Date.now = jest.genMockFunction().mockReturnValue(now)
```
Comment (nicksellen) [inline src/components/General/DateAsWords.spec.js]: Is this different to using `jest.fn()`?

**Comment by tiltec** [comment]
Looks really good, looking forward to merge this!

**Comment by nicksellen** [comment]
Related issue is https://github.com/yunity/karrot-frontend/issues/311

File: package.json (hunk 3 above)
Comment (ReneCode) [reply package.json]: oh, seems that I have to install yarn (normaly I'm working with npm)

Comment (nicksellen) [reply package.json]: Yup, we ran into a lot of issues with npm 5 (see https://github.com/yunity/karrot-frontend/issues/543)

File: src/components/General/DateAsWords.spec.js (hunk 4 above)
Comment (ReneCode) [reply src/components/General/DateAsWords.spec.js]: I've seen an example with "getMockFunction()" - but you're right: jest.fn() does the same and looks simpler.

**Pull request merged**